Source/spells.cpp

```diff
@@ -131,15 +127,13 @@ int GetManaAmount(PlayerStruct &player, spell_id sn)
 	}
 
 	if (sn == SPL_HEAL || sn == SPL_HEALOTHER) {
-		ma = (spelldata[SPL_HEAL].sManaCost + 2 * player._pLevel - adj);
+		ma = std::max((spelldata[SPL_HEAL].sManaCost + 2 * player._pLevel - adj), 0);
 	} else if (spelldata[sn].sManaCost == 255) {
-		ma = ((BYTE)player._pMaxManaBase - adj);
+		ma = std::max(((BYTE)player._pMaxManaBase - adj), 0);
 	} else {
-		ma = (spelldata[sn].sManaCost - adj);
+		ma = std::max((spelldata[sn].sManaCost - adj), 0);
 	}
 
-	if (ma < 0)

```

Comment:
why not leave the code above alone and just use max here?

Reply:
ahh you are right, updated the pull request. Thank you @qndel !